sbom-utility-scripts/scripts/index-image-sbom-script/index_image_sbom_script.py

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env python3
+import argparse
+import hashlib
+import json
+from datetime import datetime, timezone
+from pathlib import Path
+from typing import Optional, Any
+from dataclasses import dataclass
+from uuid import uuid4
+
+from packageurl import PackageURL
+
+
+@dataclass
+class Image:
+    repository: str
+    name: str
+    digest: str
+    tag: str
+    arch: Optional[str]
+
+    @staticmethod
+    def from_image_index_url_and_digest(
+        image_url_and_tag: str,
+        image_digest: str,
+        arch: Optional[str] = None,
+    ) -> "Image":
+
+        repository, tag = image_url_and_tag.rsplit(":", 1)
+        _, name = repository.rsplit("/", 1)
+        return Image(
+            repository=repository,
+            name=name,
+            digest=image_digest,
+            tag=tag,
+            arch=arch,
+        )
+
+    @property
+    def digest_algo(self) -> str:
+        algo, _ = self.digest.split(":")
+        return algo.upper()
+
+    @property
+    def digest_hex_val(self) -> str:
+        _, val = self.digest.split(":")
+        return val
+
+    def purls(self, index_digest: Optional[str] = None) -> list[str]:
+        ans = []
+        if index_digest and self.arch:
+            ans.append(
+                PackageURL(
+                    type="oci",
+                    name=self.name,
+                    version=index_digest,
+                    qualifiers={"arch": self.arch, "repository_url": self.repository},
+                ).to_string()
+            )
+        ans.append(
+            PackageURL(
+                type="oci", name=self.name, version=self.digest, qualifiers={"repository_url": self.repository}
+            ).to_string()
+        )
+        return ans
+
+    def propose_spdx_id(self) -> str:
+        purl_hex_digest = hashlib.sha256(self.purls()[0].encode()).hexdigest()
+        return f"SPDXRef-Image-{self.name}-{purl_hex_digest}"
+
+
+def create_package(image: Image, spdxid: Optional[str] = None, image_index_digest: Optional[str] = None) -> dict:
+    return {
+        "SPDXID": image.propose_spdx_id() if not spdxid else spdxid,
+        "name": image.name if not image.arch else f"{image.name}_{image.arch}",
+        "versionInfo": image.tag,
+        "supplier": "Organization: Red Hat",
```

Comment:
How much do we need the `supplier` information?

We shouldn't hardcode `Red Hat`, we can't assume only Red Hat is using Konflux. In fact, I believe Fedora is setting up / has set up their Konflux instance as well.

Can we leave it as NOASSERTION or remove it altogether?

Reply:
This is taken from ProdSec guidelines, as seen [here](https://redhatproductsecurity.github.io/security-data-guidelines/sbom/#packages-and-relationships). However I can see how this can become a problem. I will change it to `NOASSERTION`.